nixos/packetbeat: Add basic module for packetbeat

lejonet opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Basic module to enable configuring and enabling packetbeat to send
packet data to elasticsearch. Modeled after the journalbeat service.

We have the packetbeat package in nixpkgs for some time, but no module, so I decided to make a basic one. I decided to make the configuration for flows and protocols their separate config values, so that if all you want to modify is general settings and outputs, you don't have to copy the entire default config and maintain it yourself.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

MrMebelMan committed:
django-maintenance-mode: init at 0.14.0

magnetophon committed:
surge: init at 1.6.6

worldofpeace committed:
nixos/doc/releases: update the docs as promised

This goes through a recent example of 19.09 (because the workflow
should be everchanging, so our example needs to be recent).
Lots of changes, just read idk.

dasJ committed:
nixos/systemd: Don't use apply for $PATH

When not using apply, other modules can use $PATH as a list instead of
getting a colon-separated list to each /bin directory.

tfmoraes committed:
Zettlr: init at 1.7.5

Fixes  #81516

r-ryantm committed:
honggfuzz: 2.2 -> 2.3.1

tfmoraes committed:
meshlab: 2020.03 -> 2020.07

lejonet committed:
nixos/packetbeat: Add basic module for packetbeat

Basic module to enable configuring and enabling packetbeat to send
packet data to elasticsearch. Modeled after the journalbeat service.

ju1m committed:
nixos/biboumi: init

aanderse commented on nixos/modules/services/logging/packetbeat.nix:
```diff
@@ -0,0 +1,185 @@
+        '';
+      };
+
+      configFlows = mkOption {
```
If this is just `yaml` does it even need is open option? [RFC42](https://github.com/NixOS/rfcs/pull/42) has been promoting a `settings` option which I think would benefit this module greatly.

lejonet replied:
It is just yaml, this was a quick way of getting the module usable. My plan was to first get something usable, then refactor it to be smart too.

aanderse commented on nixos/modules/services/logging/packetbeat.nix:
```diff
@@ -0,0 +1,185 @@
+        '';
+      };
+
+      configProtocols = mkOption {
```
Would this benefit from a structured type? Maybe `type = with types; attrsOf (either (bool (listOf port)));`.

lejonet replied:
Not impossible, its purely yaml it needs to generate, so a structured type would probably fit. As said in other comments, this was a quick way to getting it usable.

aanderse commented on nixos/modules/services/logging/packetbeat.nix:
```diff
@@ -0,0 +1,185 @@
+let
+  cfg = config.services.packetbeat;
+
+  packetbeatYml = pkgs.writeText "packetbeat.yml" ''
```
See comment below about a `settings` option which could replace this.

lejonet replied:
I will look into that RFC, I've added the configFile option, that short-circuits the config options in the module, and lets the user handle the configuration file as they please.

aanderse reviewed (commented):
I have left some comments I hope you find useful. Please don't hesitate to ask if you need any clarification or help on points mentioned.

lejonet commented on nixos/modules/services/logging/packetbeat.nix:
```diff
@@ -0,0 +1,185 @@
+        mkdir -p ${cfg.stateDir}/data
+        mkdir -p ${cfg.stateDir}/logs
+      '';
+      serviceConfig = {
```
This is completely copied from the official unit, packetbeat needs to run as root because it captures packets from interfaces. I dunno if CAP_NET_ADMIN might be enough instead of having root, but this is how the official systemd unit does it.

lejonet commented on nixos/modules/services/logging/packetbeat.nix:
```diff
@@ -0,0 +1,185 @@
+        description = "Tags to place on the shipped log messages";
+      };
+
+      stateDir = mkOption {
```
This is 100% copied from the journalbeat module, I also wonder what value this gives, either we just hardcode or make the entire path configurable. I'm more inclined towards the latter.

lejonet replied:
I changed so that its a full path, and removed the assertion that checked for leading /

lejonet committed:
nixos/packetbeat: Make stateDir a full path instead of a path under /var/lib

lejonet committed:
nixos/packetbeat: Add configFile option

With the configFile option, the user can completely skip the configuration options
of the module and handle the configuration file themselvels.

lejonet committed:
nixos/packetbeat: Remove a trailing whitespace

aanderse committed:
nixos/packetbeat: add settings option

aanderse commented:
A bit of a rough sketch but what about something like [this](https://github.com/NixOS/nixpkgs/compare/master...aanderse:packetbeat)?

aanderse commented on nixos/modules/services/logging/packetbeat.nix:
```diff
@@ -0,0 +1,190 @@
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p ${cfg.stateDir}/data
+        mkdir -p ${cfg.stateDir}/logs
```
Is it possible to log to `stdout` instead so `journald` can pick this up?

lejonet replied:
There is an option to log to stderr (-e) that could be used in ExecStart

lejonet replied:
There is an option to log to stderr that should make it possible for journald to pick it up (-e).

lejonet committed:
nixos/packetbeat: Merge @aanderse suggestion with some fixes

lejonet commented:
> A bit of a rough sketch but what about something like [this](https://github.com/NixOS/nixpkgs/compare/master...aanderse:packetbeat)?

Well, I've merged your suggestion and fixed it up a bit, however I cannot get it to evaluate. It would seem like there is something wrong with the lib.formats.yaml.generate call, I get the following error message:
`error: value is a function while a set was expected, at /home/lejonet/sources/builddirs/lejonet-nixpkgs/nixos/modules/services/logging/packetbeat.nix:149:20`

aanderse commented:
@lejonet I'll *try* to get back to this in the next few days...

rycee committed:
openambit: init at 0.5

worldofpeace committed:
fixup! nixos/doc/releases: update the docs as promised

jtojnar committed:
fixup! nixos/doc/releases: update the docs as promised

aanderse commented:
I'm sorry this took me so long to circle back to. I've pushed another copy which actually builds now... [here](https://github.com/aanderse/nixpkgs/commit/6df023ffc3825c40999bdec27752f8ffeb70e1b4).

Maybe @Infinisil wants to take a quick look over this module and offer some helpful advice in general, and specifically for the 2 problematic lines:
```
      packetbeat = {
        interfaces.device = "any";

        # FIXME: these are the 2 problem lines
        # flows = cfg.flows;
        # protocols = cfg.protocols;
      };
```

Infinisil commented:
I had to correct the type from `either (bool str (listOf str))` to `oneOf [ boot str (listOf str) ]`, but I also did a refactoring using freeform modules as introduced in #82743, which really brings this module to the next level, see https://github.com/Infinisil/nixpkgs/commit/f3f2130cd9d555d8c09859db1f07506f81d54eb3

aanderse commented:
Those freeform module options look great! Thanks for taking a look at this module @Infinisil.

lejonet commented:
Sweet, I'll hopefully have some time this weekend to look at this and merge your changes, so that I can do the testruns of the module.

Just as a curious question, what is the actual difference between either and oneOf? Like, semantically, it would seem like they're the same, because both of them choose exactly one of a list of options, but in this case, I guess the implementations might differ in such a way that oneOf works with the freeform module suggestion and either doesn't?

Infinisil commented:
`oneOf` is just shorthand for multiple `either`s, see https://github.com/NixOS/nixpkgs/pull/65728

deviant committed:
treewide: convert patch series files to Nix expressions

Nix expressions provide all the features that Quilt series files have,
so using those instead is pointless. Also, lib.readPathsFromFile
(the function used to read series files) has the following warning:

> NOTE: This function is not performant and should be avoided.

This also removes some orphaned series files, and unused references to
copyPathsToStore (which were probably missed in previous commits
where calls to lib.readPathsFromFile were removed)

deviant committed:
lib/strings: deprecate readPathsFromFile

> NOTE: This function is not performant and should be avoided.

It's not used at all in-tree now, so we can remove it completely after
any remaining users are given notice.

rople380 committed:
godu: init at 1.3.0

Co-authored-by: Daniël de Kok <me@github.danieldk.eu>

SCOTT-HAMILTON committed:
vokosscreen-ng: init at 3.0.5

jonringer committed:
catt: fix pychromecast version

risicle committed:
pythonPackages.httpcore: init at 0.10.2

Commit:
python3Packages.phonopy: fix build and tests

risicle committed:
pythonPackages.httpx: 0.12.1 -> 0.14.2

slim down some no longer needed dependencies

risicle committed:
pythonPackages.asgi-csrf: init at 0.7

risicle committed:
pythonPackages.mergedeep: init at 1.3.0

risicle committed:
pythonPackages.starlette: 0.13.6 -> 0.13.8

remove outdated comment

risicle committed:
pythonPackages.datasette: 0.39 -> 0.46

enable more tests that now seem to work

r-ryantm committed:
python27Packages.dominate: 2.5.1 -> 2.5.2

r-ryantm committed:
python27Packages.josepy: 1.3.0 -> 1.4.0

vbgl committed:
ocamlPackages.re: disable tests with OCaml < 4.04

vbgl committed:
ocamlPackages.ocp-index: 1.2 → 1.2.1

magnetophon committed:
surge: 1.6.6 -> 1.7.1

vbgl committed:
ocamlPackages.piqi-ocaml: 0.7.5 → 0.7.7

lejonet committed:
nixos/packetbeat: Merge infinsils suggestions and add some top-level config settings

lejonet commented:
I'm getting a weird issue that I hope that you guys can help with, my nix-fu is way too rusty and basic to find the cause of this. With the module in the current state and the following config:
```
packetbeat = {
  enable = true;
  package = pkgs.packetbeat7;
  settings = {
    name = config.networking.hostName;
    packetbeat = {
      interfaces = {
        device = "ens3";
        type = "af_packet";
        ignore_outgoing = true;
      };
      protocols = {
        tls = {
          ports = [ 443 993 995 5223 8443 8883 9243 ];
          send_certificates = true;
          include_raw_certificates = false;
          include_detailed_fields = true;
          fingerprints = [ "md5" "sha1" "sha256" ];
        };
      };
    };
    output = {
      file = {
        path = "/var/log/packetbeat";
        filename = "packetbeat.log";
        rotate_every_kb = 102400;
        number_of_files = 20;
      };
    };
    fields = {
      env = "utility";
    };
    processors = [ "add_host_metadata: ~" ];
   };
 };
```

The error I get when I try to deploy the config is:
```
building all machine configurations...
error: value is a string while a set was expected, at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:262:25
(use '--show-trace' to show detailed location information)
```

The traceback isn't all to useful for me, but maybe it is for you, I think its whining about something with the ExecStart value, but that is supposed to be a string, isnt' it?
Here's the traceback:
```
building all machine configurations...
error: while evaluating the attribute 'buildCommand' of the derivation 'nixops-machines' at /home/lejonet/sources/builddirs/nixpkgs/pkgs/build-support/trivial-builders.nix:7:14:
while evaluating anonymous function at /nix/store/2piclzh8d93h9rfnwzjps4a31q0z07r4-nixops-1.7/share/nix/nixops/eval-machine-info.nix:371:46, called from undefined position:
while evaluating the attribute 'config.system.build.toplevel' at /home/lejonet/sources/builddirs/nixpkgs/nixos/modules/system/activation/top-level.nix:275:5:
while evaluating 'foldr' at /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:52:20, called from /home/lejonet/sources/builddirs/nixpkgs/nixos/modules/system/activation/top-level.nix:139:12:
while evaluating 'fold'' at /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:55:15, called from /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:59:8:
while evaluating 'showWarnings' at /home/lejonet/sources/builddirs/nixpkgs/lib/trivial.nix:302:28, called from /home/lejonet/sources/builddirs/nixpkgs/nixos/modules/system/activation/top-level.nix:136:10:
while evaluating 'foldr' at /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:52:20, called from /home/lejonet/sources/builddirs/nixpkgs/lib/trivial.nix:302:33:
while evaluating 'fold'' at /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:55:15, called from /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:59:8:
while evaluating the attribute 'warnings' at undefined position:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:84:45, called from undefined position:
while evaluating the attribute 'value' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:379:9:
while evaluating the option `warnings':
while evaluating the attribute 'mergedValue' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:411:5:
while evaluating the attribute 'values' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:405:9:
while evaluating the attribute 'values' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:505:7:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:391:28, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:391:17:
while evaluating 'dischargeProperties' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:464:25, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:392:62:
while evaluating the attribute 'value' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:277:44:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:234:10, called from undefined position:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/nixos/modules/system/boot/systemd.nix:824:51, called from /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:234:16:
while evaluating 'optional' at /home/lejonet/sources/builddirs/nixpkgs/lib/lists.nix:254:20, called from /home/lejonet/sources/builddirs/nixpkgs/nixos/modules/system/boot/systemd.nix:825:7:
while evaluating the attribute 'serviceConfig.Type' at undefined position:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:84:45, called from undefined position:
while evaluating the attribute 'value' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:379:9:
while evaluating the option `systemd.services.packetbeat.serviceConfig':
while evaluating the attribute 'mergedValue' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:411:5:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:413:22, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:413:9:
while evaluating 'merge' at /home/lejonet/sources/builddirs/nixpkgs/lib/types.nix:292:20, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:416:12:
while evaluating 'filterAttrs' at /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:124:23, called from /home/lejonet/sources/builddirs/nixpkgs/lib/types.nix:293:35:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:125:29, called from /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:125:18:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/types.nix:293:51, called from /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:125:62:
while evaluating the attribute 'ExecStart' at /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:344:7:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/types.nix:293:86, called from /home/lejonet/sources/builddirs/nixpkgs/lib/attrsets.nix:344:15:
while evaluating the attribute 'optionalValue' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:424:5:
while evaluating the attribute 'values' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:405:9:
while evaluating the attribute 'values' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:505:7:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:391:28, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:391:17:
while evaluating 'dischargeProperties' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:464:25, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:392:62:
while evaluating the attribute 'value' at /home/lejonet/sources/builddirs/nixpkgs/lib/types.nix:297:60:
while evaluating the attribute 'value' of the derivation 'packetbeat.yml' at /home/lejonet/sources/builddirs/nixpkgs/pkgs/build-support/trivial-builders.nix:7:14:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:84:45, called from undefined position:
while evaluating 'isType' at /home/lejonet/sources/builddirs/nixpkgs/lib/types.nix:14:18, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:85:14:
while evaluating the attribute '_type' at undefined position:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:280:39, called from undefined position:
while evaluating 'mergeModules'' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:237:36, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:298:13:
while evaluating 'flip' at /home/lejonet/sources/builddirs/nixpkgs/lib/trivial.nix:138:16, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:280:6:
while evaluating 'byName' at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:260:25, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:268:21:
while evaluating anonymous function at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:261:22, called from /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:261:9:
value is a string while a set was expected, at /home/lejonet/sources/builddirs/nixpkgs/lib/modules.nix:262:25
```

orivej committed:
linux-rt: add update script

orivej committed:
linux-rt_5_4: init at 5.4.61-rt37

orivej committed:
linux-rt_5_6: init at 5.6.19-rt12

zowoq committed:
dep2nix: regenerate deps.nix

zowoq committed:
leaps: regenerate deps.nix

lovek323 committed:
goreplay: init at 1.1.0

r-ryantm committed:
zenmonitor: 1.4.1 -> 1.4.2

r-ryantm committed:
yaru-theme: 20.10.1 -> 20.10.2

saschagrunert committed:
nixos/containers: add oci-seccomp-bpf-hook

Signed-off-by: Sascha Grunert <sgrunert@suse.com>

rople380 committed:
maintainers: add rople380

Gerschtli committed:
rustup: add zlib as runtime dependency
Fixes https://github.com/NixOS/nixpkgs/issues/92946.

dotlambda committed:
radicale: 3.0.3 -> 3.0.5

drewrisinger committed:
python3Packages.cirq: unbreak

Remove unneeded test disables.
Add changelog.
Unpin most python dependencies.

marsam committed:
ocaml-ng.ocamlPackages_4_11.ocaml: 4.11.0 -> 4.11.1

dotlambda committed:
etesync-dav: loosen radicale version bounds

zowoq committed:
oci-seccomp-bpf-hook: move to linuxPackages

jonringer committed:
python39: 3.9.0b5 -> 3.9.0rc1

davidak committed:
limesurvey: 3.17.12+190823 -> 3.23.0+200813

fixes at least 17 CVEs
also adds test to package

zowoq committed:
go_1_14: 1.14.7 -> 1.14.8

jtojnar committed:
strigi: drop

It has not been used by KDE for many years and depends on umaintained libraries we want to drop (Qt4 and Gamin).

jtojnar committed:
Merge pull request #96120 from jtojnar/cleaner-wx

jtojnar committed:
gstreamer: drop legacy

It has been abandoned for ages, insecure, and now finally broken.

jtojnar committed:
qt4: drop multimedia & webkit options

They have been broken and disabled for ages and now the dependencies are being removed.

jtojnar committed:
steam-run: remove ancient gstreamer dependency

It is insecure and broken.

Follow https://github.com/NixOS/nixpkgs/pull/93398

jtojnar committed:
appimage: remove ancient gst dep

it is insecure and broken

colemickens committed:
cordless: init at 2020-08-30

jtojnar committed:
gst_all_1.gst-validate: use correct gstreamer

previously it was using the legacy one

jtojnar committed:
googleearth: do not use legacy gst

jtojnar committed:
webcamoid: drop legacy gst

marsam committed:
Merge pull request #96905 from marsam/update-ncspot

ncspot: 0.2.1 -> 0.2.2

ttuegel committed:
Merge pull request #84542 from petabyteboy/feature/plasma-5-18

plasma5: 5.17.5 -> 5.18.5

marsam committed:
ibm-plex: 5.1.0 -> 5.1.3

marsam committed:
Merge pull request #96918 from lucperkins/doppler-3.10.3

doppler: 3.10.1 -> 3.10.3

jtojnar committed:
tiscamera: drop legacy gst

Commit:
git-town: 7.3.0 → 7.4.0

jtojnar committed:
tiscamera: clean up

- use pkg-config instead of pkgconfig alias
- use placeholders for cmakeFlags
- do not skip build rpath
- udev replacements no longer apply https://github.com/TheImagingSource/tiscamera/commit/47d4fb246b5f82763f87ad57c7d1ed3420655e96

There are some misplaced dependencies like gobject-introspection and orc that should probably go to nativeBuildInputs but upstream detects them incorrectly.

Ericson2314 committed:
Merge pull request #61019 from volth/gcc.arch-amd

platform.gcc.arch: support for AMD CPUs

orivej committed:
tunefish: init at unstable 2020-08-13 (#96539)

Fails to build on aarch64 due to the use of x86 intrinsics.

orivej-nixos committed:
Merge pull request #96702 from orivej/linux-rt

linux-rt(,_latest): init at 5.4.61-rt37 and 5.6.19-rt12

orivej committed:
nixos/systemd: support adding and overriding tmpfiles.d via environment.etc (#96766)

This allows the user to configure systemd tmpfiles.d via
`environment.etc."tmpfiles.d/X.conf".text = "..."`, which after #93073
causes permission denied (with new X.conf):

```
ln: failed to create symbolic link '/nix/store/...-etc/etc/tmpfiles.d/X.conf': Permission denied
builder for '/nix/store/...-etc.drv' failed with exit code 1
```

or collision between environment.etc and systemd-default-tmpfiles
packages (with existing X.conf, such as tmp.conf):

```
duplicate entry tmpfiles.d/tmp.conf -> /nix/store/...-etc-tmp.conf
mismatched duplicate entry /nix/store/...-systemd-246/example/tmpfiles.d/tmp.conf <-> /nix/store/...-etc-tmp.conf
builder for '/nix/store/...-etc.drv' failed with exit code 1
```

Fixes #96755

orivej committed:
surge: fix build

marsam committed:
git-lfs: 2.11.0 -> 2.12.0

https://github.com/git-lfs/git-lfs/releases/tag/v2.12.0

jtojnar committed:
pitivi: clean up

* format with nixpkgs-fmt
* use pkg-config instead of pkgconfig alias
* reorder attributes
* move version inside the mkDerivation

endgame committed:
chromedriver: 83.0.4103.39 -> 85.0.4183.87

New dependency: libxcb

orivej-nixos committed:
Merge pull request #82308 from magnetophon/surge

surge: init at 1.7.1

orivej-nixos committed:
Merge pull request #96684 from magnetophon/LV2plugins

update some LV2 plugins

jtojnar committed:
pitivi: Fix build

The latest version is not compatible with Python 3.8,
let's cherry-pick some patches from master.

Also pre-prepare for incoming release of GStreamer 1.18.

jtojnar committed:
pkgsMusl.libusb-compat-0_1: fix eval

e89e2edc7349c0df15fa03b8229c370340547538 did not move all files when moving the directory.

Fixes: https://github.com/NixOS/nixpkgs/issues/96947

danieldk committed:
Merge pull request #96904 from danieldk/1password-0.8.4

_1password-gui: 0.8.3-1 -> 0.8.4

danieldk committed:
Merge pull request #96944 from xoe-labs/da-bump-gittown

git-town: 7.3.0 → 7.4.0

worldofpeace committed:
Merge pull request #96879 from romildo/rm.deepin.doc

deepin: register removal in release notes, aliases and module list

danieldk committed:
Merge pull request #96282 from rople380/master

godu: init at 1.3.0

vcunat committed:
powerdevil: fix hash of patch

The hash added in 24ebfb1e021b belongs to pure download
before normalization done by fetchpatch.

SCOTT-HAMILTON committed:
numworks-udev-rules: init on 02/09/2020

marsam committed:
Merge pull request #96945 from endgame/chromedriver-bump

chromedriver: 83.0.4103.39 -> 85.0.4183.87

worldofpeace committed:
Merge pull request #96042 from rnhmjoj/loaOf

treewide: completely remove types.loaOf

worldofpeace committed:
Merge pull request #96937 from jtojnar/drop-strigi

strigi: drop

victorluft committed:
sublime-merge-dev: 2022 -> 2031

victorluft committed:
sublime-merge: 2020 -> 2033

elseym committed:
Merge pull request #96677 from edude03/patch-2

plex: 1.19 -> 1.20

aanderse committed:
Merge pull request #96738 from jojosch/dbeaver_7.2.0

dbeaver: 7.1.5 -> 7.2.0

Commit:
Merge pull request #94917 from ju1m/biboumi

nixos/biboumi: init

aanderse committed:
Merge pull request #96926 from davidak/limesurvey-update

limesurvey: 3.17.12+190823 -> 3.23.0+200813

symphorien committed:
python3Packages.matrix-nio: 0.14.1 -> 0.15.1

https://github.com/poljar/matrix-nio/blob/master/CHANGELOG.md

no obvious compatibility issue

symphorien committed:
pantalaimon: 0.6.5 -> 0.7.0

changelog: https://github.com/matrix-org/pantalaimon/blob/0.7.0/CHANGELOG.md

Commit:
gitAndTools.gita: 0.10.9 -> 0.10.10

NeQuissimus committed:
oh-my-zsh: 2020-08-31 -> 2020-09-02

danieldk committed:
Merge pull request #96831 from SCOTT-HAMILTON/numworks

numworks: init udev rules and module

jtojnar committed:
Merge pull request #96948 from jtojnar/fix-pitivi

worldofpeace committed:
Merge pull request #96925 from jtojnar/gpaste-session-path

nixos/gpaste: return sessionPath

dasJ committed:
left4gore: Init at 2.3

Ericson2314 committed:
dash: Fix cross build

jonringer committed:
python3Packages.azure-mgmt-synapse: init at 0.3.0

jonringer committed:
python3Packages.azure-multiapi-storage: 0.3.5 -> 0.4.1

jonringer committed:
python3Packages.azure-mgmt-appconfiguration: 0.5.0 -> 0.6.0

jonringer committed:
python3Packages.azure-mgmt-containerservice: 9.2.0 -> 9.3.0

jonringer committed:
python3Packages.azure-mgmt-reservations: 0.7.0 -> 0.8.0

jonringer committed:
python3Packages.azure-mgmt-storage: 11.1.0 -> 11.2.0

primeos committed:
signal-desktop: 1.34.5 -> 1.35.1

jonringer committed:
python3Packages.azure-synapse-accesscontrol: init at 0.2.0

oyren committed:
portfolio: 0.48.0 -> 0.48.1

jonringer committed:
python3Packages.azure-synapse-spark: init at 0.2.0

jonringer committed:
python3Packages.azure-mgmt-keyvault: fix overridden package

jonringer committed:
azure-cli: 2.10.1 -> 2.11.1

risicle committed:
Merge pull request #96500 from r-ryantm/auto-update/python2.7-elasticsearch

python27Packages.elasticsearch: 7.8.1 -> 7.9.1

marsam committed:
rclone: 1.52.3 -> 1.53.0

matthewbauer committed:
Merge pull request #96563 from obsidiansystems/skip-cudnn_cnn_infer-instead-of-removing

cudnn: skip libcudnn_cnn_infer.so, instead of removing

matthewbauer committed:
Merge pull request #96402 from matthewbauer/bazel-jdk-headless

Fix Bazel on macOS

matthewbauer committed:
Merge pull request #96562 from obsidiansystems/bazel-add-dont-add-bazel-opts

build-bazel-package: add dontAddBazelOpts arg

jperras committed:
black-macchiato: init at 1.3.0

Adds the Python package `black-macchiato`, which allows the usage of
`black` for subsets of an entire file.

risicle committed:
Merge pull request #96310 from yvesf/qcad-3.25.0.0

update qcad to 3.25.0.0

risicle committed:
Merge pull request #96561 from jonringer/fix-jupyterhub-ldapauthenticator

python3Packages.jupyterhub-ldapauthenticator: 1.3.1 -> 1.3.2, fix build

jperras committed:
maintainers: add jperras

marsam committed:
ocamlPackages.csexp: init at 1.3.1

marsam committed:
dune_2: 2.7.0 -> 2.7.1

https://github.com/ocaml/dune/releases/tag/2.7.1

marsam committed:
Merge pull request #96943 from marsam/update-ibm-plex

ibm-plex: 5.1.0 -> 5.1.3

Commit:
terraform_0_13: 0.13.1 -> 0.13.2 (#96971)

risicle committed:
Merge pull request #96501 from jonringer/fix-phonopy

python3Packages.phonopy: fix build and tests

trepetti committed:
svlint: init at 0.4.7

Commit:
vimPlugins: Remove coc-* from vim-plugin-names

Because these plugins use the node packages as their source, there's no
reason to pull their sources from GitHub only to have it be overriden.

This will trigger the removal of these plugins from generated.nix the
next time update.py is run.

Commit:
vimPlugins: Refactor coc-* plugins and fix coc-go

There are 32 vim plugins, all related to the coc.nvim LSP plugin, that
are actually node packages and are included in nodePackages. In #82578,
those vim plugins were made to point at the corresponding node packages,
which fixed issues where the sources weren't being compiled properly
with Javascript build systems.

The way each of those vim plugins wraps its corresponding node package
is identical, so I factored that out into a function and eliminated the
duplication.

In addition, for some reason coc-go got missed, which led to it not
working correctly for me. I included it in my list of packages to wrap,
so that fixes that issue.

Commit:
vimPlugins: update

marsam committed:
recursive: 1.062 -> 1.064

marsam committed:
ocamlPackages.merlin: 3.3.8 -> 3.3.9

https://github.com/ocaml/merlin/releases/tag/v3.3.9

marsam committed:
flow: 0.132.0 -> 0.133.0

https://github.com/facebook/flow/releases/tag/v0.133.0

equirosa committed:
croc: 8.0.13 -> 8.3.0

7c6f434c committed:
Merge pull request #96474 from dguibert/dg/libreoffice-7

libreoffice-fresh: 6.4.3.2 -> 7.0.0.3

7c6f434c committed:
Merge pull request #96494 from jitwit/j

j: j901e -> j901f

Mic92 committed:
Merge pull request #89748 from heinic/krb5-lists

danieldk committed:
programs.zsh: remove unnecessary `with`

rnhmjoj committed:
Merge pull request #96997 from danieldk/oh-my-zsh-unnecessary-with-pkgs

programs.zsh: remove unnecessary `with`

Ma27 committed:
bandwhich: 0.16.0 -> 0.17.0

https://github.com/imsnif/bandwhich/releases/tag/0.17.0

Ma27 committed:
element-web: 1.7.4 -> 1.7.5

https://github.com/vector-im/element-web/releases/tag/v1.7.5

Ma27 committed:
element-desktop: 1.7.4 -> 1.7.5

https://github.com/vector-im/element-desktop/releases/tag/v1.7.5

doronbehar committed:
gst_all_1.gst-rtsp-server: fix pkgconfig includedir and libdir

timokau committed:
fpc: 3.0.4 -> 3.2.0

There was no 3.1 for some reason. The old sed-based path patching was
broken and resulted in syntax errors since it was a bit over-eager.
Instead of fixing it, I decided to replace it with a patch file which is
easier to inspect and will fail in a more obvious way next time.

The patch is now applied unconditionally, since it actually applies to
all linux platforms. The changes are localized to linux-specific code,
so it does not hurt to apply it on non-linux platforms as well.

Hedgewars needs a small fix to work with the new version. Done in the
same commit to avoid a broken commit.

timokau committed:
fpc: add support for aarch64-linux

Supported since fpc 3.2.0.

Commit:
Add vscode json language server

Commit:
Update node packages nix expression

samuelgrf committed:
kwin-dynamic-workspaces: init at 1.0.1

B4dM4n committed:
libreoffice-still: fix build

timokau committed:
ultrastardx: unstable-2019-01-07 -> 2020.4.0

Might as well update it while I fix it up for the fpc update.

mweinelt committed:
Merge pull request #96595 from mweinelt/pinnwand

pinnwand: 1.2.1 -> 1.2.2

timokau committed:
ddrescueview: 0.4alpha3 -> 0.4alpha4

Update required for compatibility with fpc 3.2.0.

Lassulus committed:
Merge pull request #86663 from helsinki-systems/init-left4gore

left4gore: Init at 2.3

timokau committed:
hedgewars: fix build on aarch64

Possible now that fpc works on aarch64. For some reason hedgewars
requires llvm when building on that platform.

dguibert committed:
Merge pull request #97015 from B4dM4n/libreoffice-still-fix-build

libreoffice-still: fix build

samuelgrf committed:
krohnkite: fix typo

samuelgrf committed:
kwin-tiling: fix typo

trepetti committed:
svls: init at 0.1.17

Commit:
securefs: 0.8.3 -> 0.11.1

DamienCassou committed:
emacs: Add missing patches to the Emacs derivations

patches were removed by a recent refactoring of Emacs packages:

  967259e6b49 * emacs: Factor out expression to a generic build

zimbatm committed:
direnv: 2.21.3 -> 2.22.0 (#97011)

roberth committed:
Merge pull request #95760 from janders223/master

Add vscode-json-languageserver-bin node package

infinisil committed:
Merge pull request #96564 from K900/murmur-grpc

murmur: support building with grpc

Infinisil commented:
`freeformType` shouldn't be within the option definitions, so do this to fix it:
```patch
diff --git a/nixos/modules/services/logging/packetbeat.nix b/nixos/modules/services/logging/packetbeat.nix
index 97f4c03944f..4a32cc27213 100644
--- a/nixos/modules/services/logging/packetbeat.nix
+++ b/nixos/modules/services/logging/packetbeat.nix
@@ -35,8 +35,8 @@ in
 
       settings = mkOption {
         type = types.submodule {
+          freeformType = format.type;
           options = {
-            freeformType = format.type;
             name = mkOption {
               type = types.str;
               default = "packetbeat";
```

You also need to rebase your PR on master again, because in the current version `freeformType` isn't implemented yet

cmacrae committed:
yabai: 3.2.1 -> 3.3.0

jonringer committed:
python3Packages.ftputil: fix tests

timokau committed:
Merge pull request #97006 from timokau/fpc-3.2.0-aarch64

fpc: 3.0.4 -> 3.2.0, add support for aarch64-linux

NeQuissimus committed:
linux: 4.14.195 -> 4.14.196

Izorkin committed:
htop: 3.0.0 -> 3.0.1

NeQuissimus committed:
linux: 4.19.142 -> 4.19.143

NeQuissimus committed:
linux: 4.4.234 -> 4.4.235

NeQuissimus committed:
linux: 4.9.234 -> 4.9.235

NeQuissimus committed:
linux: 5.4.61 -> 5.4.62

NeQuissimus committed:
linux: 5.8.5 -> 5.8.6

FRidh committed:
pythonPackages.dnspython: use 1.x for Python 2

Ma27 committed:
Merge pull request #97032 from Izorkin/htop

htop: 3.0.0 -> 3.0.1

FRidh committed:
pythonPackages.dnspython_1: add missing file, fixes eval

deviant committed:
breeze-plymouth: remove trailing whitespace

jbaum98 committed:
vimPlugins: Document node-based plugins

Ericson2314 committed:
Merge pull request #96970 from obsidiansystems/dash-cross

dash: Fix cross build

r-ryantm committed:
welle-io: 2.1 -> 2.2

saschagrunert committed:
buildah: 1.15.1 -> 1.15.2

Signed-off-by: Sascha Grunert <sgrunert@suse.com>

Mic92 committed:
Merge pull request #97031 from timokau/hedgewars-aarch64

hedgewars: fix build on aarch64

danieldk committed:
Merge pull request #96900 from danieldk/zotero-impurity

zotero: fix reliance on coreutils being in PATH

adisbladis committed:
Merge pull request #97026 from DamienCassou/add-patches-back-to-emacs

emacs: Add missing patches to the Emacs derivations

markuskowa committed:
openmpi: 4.0.4 -> 4.0.5

AndersonTorres committed:
Merge pull request #96932 from chkno/gtg-tests

gtg: Run tests

AndersonTorres committed:
Merge pull request #93631 from danielfullmer/k2pdfopt-2.52

k2pdfopt: 2.51a -> 2.53

Mic92 committed:
Merge pull request #96523 from Izorkin/netdata

Mic92 committed:
Merge pull request #96567 from kampka/gocryptfs

gocryptfs: 1.7.1 -> 1.8.0

jtojnar committed:
netpbm: clean up

reorder attributes and use pkg-config instead of the alias

Ma27 committed:
hydra-unstable: 2020-08-04 -> 2020-09-02

Needed to fix Hydra with latest `nixUnstable`.

mweinelt committed:
Merge pull request #96863 from mweinelt/zigbee2mqtt

zigbee2mqtt: 1.14.2 -> 1.14.3

infinisil committed:
Merge pull request #96269 from deviant/remove-quilt-series

treewide: convert patch series files to Nix expressions

jonringer committed:
Merge pull request #95983 from OPNA2608/update-bambootracker-0.4.4

bambootracker: 0.4.3 -> 0.4.4

risicle committed:
honggfuzz: patch hfuzz-cc.c to fix path to buildtime-provided clang

i judged this slightly better than using a wrapper

jonringer committed:
Merge pull request #96940 from colemickens/cordless

cordless: init at 2020-08-30

worldofpeace committed:
Merge pull request #97046 from Ma27/hydra-fix

hydra-unstable: 2020-08-04 -> 2020-09-02

r-ryantm committed:
yarn: 1.22.4 -> 1.22.5

ianmjones committed:
snippetpixie: 1.3.3 -> 1.4.1

peterhoeg committed:
kde applications: 20.08.0 -> 20.08.1

peterhoeg committed:
Merge pull request #97022 from samuelgrf/fix/kwin-scripts-typo

krohnkite,kwin-tiling: fix typo

infinisil committed:
Merge pull request #91092 from helsinki-systems/systemd-path-noapply

nixos/systemd: Don't use apply for $PATH

worldofpeace committed:
Merge pull request #97057 from ianmjones/fix/snippetpixie-update

snippetpixie: 1.3.3 -> 1.4.1

r-ryantm committed:
AusweisApp2: 1.20.1 -> 1.20.2

marsam committed:
Merge pull request #96990 from marsam/update-recursive

recursive: 1.062 -> 1.064

r-ryantm committed:
amfora: 1.4.0 -> 1.5.0

r-ryantm committed:
antibody: 6.1.0 -> 6.1.1

r-ryantm committed:
argo: 2.10.0 -> 2.10.1

r-ryantm committed:
asls: 0.4.2 -> 0.5.0

marsam committed:
Merge pull request #97029 from cmacrae/yabai/upgrade/3.3.0

yabai: 3.2.1 -> 3.3.0

marsam committed:
Merge pull request #96989 from marsam/update-rclone

rclone: 1.52.3 -> 1.53.0

r-ryantm committed:
bazarr: 0.9 -> 0.9.0.2

marsam committed:
Merge pull request #97059 from r-ryantm/auto-update/yarn

yarn: 1.22.4 -> 1.22.5

marsam committed:
Merge pull request #97069 from r-ryantm/auto-update/argo

argo: 2.10.0 -> 2.10.1

marsam committed:
Merge pull request #97024 from sephii/securefs_0_11_1

securefs: 0.8.3 -> 0.11.1

marsam committed:
Merge pull request #97010 from marsam/update-git-lfs

git-lfs: 2.11.0 -> 2.12.0

mjlbach committed:
pythonPackages.apsw: 3.32.2-r1 -> 3.33.0-r1

marsam committed:
Merge pull request #97012 from marsam/update-flow

flow: 0.132.0 -> 0.133.0

bbigras committed:
hotspot: 1.2.0 -> 1.3.0

markuskowa committed:
Merge pull request #97043 from markuskowa/upd-openmpi

openmpi: 4.0.4 -> 4.0.5

timokau committed:
hedgewars: enable build on hydra

Was disabled when the package was marked as broken in
c33386195407e13d588ed4e13925c78d524ba86f. It has since been un-broken,
but the hydraPlatforms exception was left presumably by mistake.

r-ryantm committed:
dapper: 0.5.1 -> 0.5.3

r-ryantm committed:
dcrd: 1.5.1 -> 1.5.2

r-ryantm committed:
dnsproxy: 0.31.0 -> 0.32.0

r-ryantm committed:
du-dust: 0.5.2 -> 0.5.3

r-ryantm committed:
espanso: 0.7.0 -> 0.7.1

r-ryantm committed:
codeql: 2.2.4 -> 2.2.5 (#97097)

NeQuissimus committed:
oh-my-zsh: 2020-09-02 -> 2020-09-03

nh2 committed:
Merge pull request #97084 from bbigras/hotspot

hotspot: 1.2.0 -> 1.3.0

ttuegel committed:
Merge pull request #97020 from samuelgrf/initpkg/dynamic-workspaces

kwin-dynamic-workspaces: init at 1.0.1

ttuegel committed:
Merge pull request #97121 from peterhoeg/u/kapps_20081

kdeApplications: 20.08.0 -> 20.08.1

timokau committed:
Merge pull request #97095 from timokau/hedgewars-hydra

hedgewars: enable build on hydra

jtojnar committed:
Merge pull request #96939 from jtojnar/drop-gst-010

jonringer committed:
Merge pull request #96853 from r-ryantm/auto-update/antimicroX

antimicroX: 3.0 -> 3.0.1

jonringer committed:
python3Packages.sqlalchemy-citext: fix tests

lejonet committed:
nixos/packetbeat: Move freeformType attribute to the right level

romildo committed:
Merge pull request #96559 from romildo/fix.efl

enlightenment.efl: add needed libraries

romildo committed:
Merge pull request #96558 from romildo/fix.terminology

enlightenment.terminology: use absolute path to edje_cc

romildo committed:
Merge pull request #96676 from romildo/upd.marwaita-manjaro

marwaita-manjaro: init at 2020-08-29

risicle committed:
Merge pull request #94579 from tfmoraes/meshlab-20.07

meshlab: 2020.03 -> 2020.07

lejonet committed:
Merge

lejonet committed:
nixos/packetbeat: Fix rebase screw up
